libraries/WiFi/examples/WiFiProv/WiFiProv.ino

```diff
@@ -0,0 +1,85 @@
+#include "WiFi.h"
+#include "wifi_provisioning/scheme_ble.h"
+static esp_err_t WiFiEvent(void *ctx,system_event_t *event)
+{
+    switch (event->event_id) {
+      case SYSTEM_EVENT_STA_START:
+          esp_wifi_connect();
+          break;
+      case SYSTEM_EVENT_STA_GOT_IP:
+          Serial.print("\nConnected with IP address : ");
+          Serial.println(ip4addr_ntoa(&event->event_info.got_ip.ip_info.ip));
+          break;
+      case SYSTEM_EVENT_STA_DISCONNECTED:
+          Serial.println("\nDisconnected. Connecting to the AP again... ");
+          esp_wifi_connect();
+          break;
+      default:
+          break;
+      }
+    return ESP_OK;
+}
+static void prov_event_handler(void *user_data, wifi_prov_cb_event_t event, void *event_data)
+{
+    switch (event) {
+        
+        case WIFI_PROV_START:
+            Serial.println("\nProvisioning started\nGive Credentials of your access point using \" Android app \"");
+            break;
+        case WIFI_PROV_CRED_RECV: { 
+            Serial.println("\nReceived Wi-Fi credentials");
+            wifi_sta_config_t *wifi_sta_cfg = (wifi_sta_config_t *)event_data;
+            Serial.print("\tSSID : ");
+            Serial.println((const char *) wifi_sta_cfg->ssid);
+            Serial.print("\tPassword : ");
+            Serial.println((char const *) wifi_sta_cfg->password);
+            break;
+        }
+        case WIFI_PROV_CRED_FAIL: { 
+            wifi_prov_sta_fail_reason_t *reason = (wifi_prov_sta_fail_reason_t *)event_data;
+            Serial.println("\nProvisioning failed!\nPlease reset to factory and retry provisioning\n");
+            if(*reason == WIFI_PROV_STA_AUTH_ERROR)
+            {
+                Serial.println("\nWi-Fi AP password incorrect");
+            }
+            else
+            {
+                Serial.println("\nWi-Fi AP not found....Add API \" nvs_flash_erase() \" before beginProvision()");
+            }           
+            break;
+        }
+        case WIFI_PROV_CRED_SUCCESS:
+            Serial.println("\nProvisioning Successful");
+            break;
+        case WIFI_PROV_END:
+            Serial.println("\nProvisioning Ends");
+            wifi_prov_mgr_deinit();
+            break;
+        default:
+            break;
+    }
+}
+
+void setup() {
+
+  
+  prov_config_t conf;
+
+  conf.prov_scheme = WIFI_PROV_SCHEME_SOFTAP;
+  conf.security = WIFI_PROV_SECURITY_1;
+  conf.scheme_event_handler = WIFI_PROV_EVENT_HANDLER_NONE;
+  conf.pop = "abcd1234";
+  conf.service_name = "WIFI_XXXX";
+  conf.service_key = "123456789"; 
+
+  Serial.begin(115200);
+  nvs_flash_init();
+  tcpip_adapter_init();
```

Comment:
this should also not be called here. it is again a part of the IP stack init process [here](https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/src/WiFiGeneric.cpp#L101). The function is not static, so you can declare and call it from your code.